src/webgl/p5.Camera.js

```diff
@@ -0,0 +1,953 @@
+/**
+ * @module Lights, Camera
+ * @submodule Camera
+ * @requires core
+ */
+
+'use strict';
+
+var p5 = require('../core/core');
+
+/**
+ * Creates a new <a href="#/p5.Camera">p5.Camera</a> object and tells the
+ * renderer to use that camera.
+ * Returns the p5.Camera object.
+ * @method createCamera
+ * @return {p5.Camera} The newly created camera object.
+ * @for p5
+ */
+p5.prototype.createCamera = function() {
+  this._assert3d('createCamera');
+  var _cam = new p5.Camera(this._renderer);
+
+  // compute default camera settings, then set a default camera
+  _cam._computeCameraDefaultSettings();
+  _cam._setDefaultCamera();
+
+  // set renderer current camera to the new camera
+  this._renderer._curCamera = _cam;
+
+  return _cam;
+};
+
+/**
+ * This P5.Camera class describes a 3D camera for use with the WebGL Renderer.
+ *
+ * Note:
+ * The methods below operate in two coordinate systems: the 'world' coordinate
+ * system describe positions in terms of their relationship to the origin along
+ * the X, Y and Z axes whereas the camera's 'local' coordinate system
+ * describes positions from the camera's point of view: left-right, up-down,
+ * and forward-backward. The <a href="#/p5.Camera/move">move()</a> method,
+ * for instance, moves the camera along its own axes, whereas the
+ * <a href="#/p5.Camera/setPosition">setPosition()</a>
+ * method sets the camera's position in world-space.
+ *
+ *
+ * @class p5.Camera
+ * @constructor
+ * @param {p5.RendererGL} renderer an instance of p5.RendererGL
+ */
+p5.Camera = function(renderer) {
+  this._renderer = renderer;
+
+  this.cameraType = 'default';
+
+  this.cameraMatrix = new p5.Matrix();
+  this.projMatrix = new p5.Matrix();
```

Comment:
The reason I kept the '_computeCameraDefaultSettings' and 'setDefaultCamera' out of the constructor was so that copying a camera wouldn't unnecessarily compute these defaults when they were going to be over-written anyway.  I figured most users would be using createCamera which calls those methods.  That said, I think there is a more elegant way to deal with defaults in general (besides storing default values and having a 'default' vs. 'custom' flag), so I'll look into that.